### PR TITLE
[nix] Rematerialize GHC9 in the CI

### PR DESCRIPTION
**Pull request by goodlyrottenapple**
Updated the `Update/ Nix` CI script to re-materialize both the current GHC8.10.* and the GHC9 expressions in the `nix/` folder.

### CODE REVIEW

File: .github/workflows/update.yml [hunk 1]
```diff
@@ -27,22 +27,26 @@ jobs:
...
         uses: cachix/cachix-action@v10
         with:
           name: runtimeverification
+          extraPullNames: kore
           signingKey: '${{ secrets.CACHIX_SIGNING_KEY }}'
 
+
       - name: Materialize
-        run: ./nix/rematerialize.sh
+        run: GC_DONT_GC=1 nix run .#update-cabal
```
Comment (jberthold) [inline .github/workflows/update.yml:46]: Should we delete the old script then?
But maybe it is useful to keep it so one can run the update locally. Then we would want to have the `nix run ...` commands in that script.

File: flake.lock [hunk 2]
```diff
@@ -98,6 +98,21 @@
         "type": "github"
       }
     },
+    "flake-utils_2": {
+      "locked": {
+        "lastModified": 1642700792,
+        "narHash": "sha256-XqHrk7hFb+zBvRg6Ghl+AZDq03ov6OshJLiSWOoX5es=",
```
Comment (jberthold) [inline flake.lock:104]: Changes in this file are the same as on my PR #3143, which might cause conflicts if git is not smart enough.

**Review by jberthold — COMMENTED** [review]
Changesets of the `kore.nix` files are a bit hard to read because of the different layout, but all makes sense.
Besides maintaining the `nix/rematerialize.sh` script, LGTM

File: .github/workflows/update.yml (hunk 1 above)
Comment (goodlyrottenapple) [reply .github/workflows/update.yml:46]: No, the old script works with nix <2.4 without flakes. It should be removed if/when we get rid of non flake nix stuff in the repo

File: flake.lock (hunk 2 above)
Comment (goodlyrottenapple) [reply flake.lock:104]: Yeah sorry I think I forgot to commit the lock file last time I modified the flake.nix file. I will merge master again if you get your changes in first

**Comment by goodlyrottenapple** [comment]
> Changesets of the `kore.nix` files are a bit hard to read because of the different layout, but all makes sense.
> Besides maintaining the `nix/rematerialize.sh` script, LGTM

Yeah sorry about that. I added a step where I run nixfmt on the generated file because for some bizzare reason, running the rematerialize on my local machine and the ci produced files with different formatting. This way we make doubly sure we only trigger the update if the contents actually change

File: .github/workflows/update.yml (hunk 1 above)
Comment (jberthold) [reply .github/workflows/update.yml:46]: :man_shrugging: OK then we keep it as it is for now.

**Review by jberthold — APPROVED** [review]

**Pull request merged**